src/Composer/Util/GitHub.php

```diff
@@ -101,9 +101,10 @@ public function authorizeOAuthInteractively($originUrl, $message = null)
         $this->io->writeError(sprintf('Tokens will be stored in plain text in "%s" for future use by Composer.', $this->config->getAuthConfigSource()->getName()));
         $this->io->writeError('For additional information, check https://getcomposer.org/doc/articles/authentication-for-private-packages.md#github-oauth');
 
-        $token = trim($this->io->askAndHideAnswer('Token (hidden): '));
+        $token = $this->io->askAndHideAnswer('Token (hidden): ');
+        $token = is_string($token) ? trim($token) : '';
```

Comment:
IMO casting is fine here, it's exactly the same effect but reads easier (to me anyway..)

```suggestion
        $token = trim((string) $this->io->askAndHideAnswer('Token (hidden): '));
```